warp10/src/main/java/io/warp10/script/binary/MACROBINARYOP.java

```diff
@@ -0,0 +1,125 @@
+
+    output = new ArgumentsBuilder()
+      .addArgument(GeoTimeSerie.class, OUTPUT, "The resulting GTS.")
+      .build();
+
+    //
+    // Equality assert unit test
+    //
+
+    StringBuilder equalTest = new StringBuilder();
+    equalTest.append("[ 1 100 <% %> FOR ] [] [] [] [ 1 100 <% RAND %> FOR ] MAKEGTS 'a' STORE" + System.lineSeparator());
+    equalTest.append("$a $a <% == ASSERT NULL %> " + getName() + " DROP");
+    getUnitTests().add(equalTest.toString());
+
+    //
+    // Simple unit test (check that results is the same than already defined binary ops)
+    //
+
+    StringBuilder addTest = new StringBuilder();
+    addTest.append("[ 1 100 <% %> FOR ] [] [] [] [ 1 100 <% RAND %> FOR ] MAKEGTS 'a' STORE" + System.lineSeparator());
+    addTest.append("[ 1 100 <% %> FOR ] [] [] [] [ 1 100 <% RAND %> FOR ] MAKEGTS 'b' STORE" + System.lineSeparator());
+    addTest.append("$a $b <% + %> " + getName() + System.lineSeparator());
+    addTest.append("$a $b + " + System.lineSeparator());
+    addTest.append("<% == ASSERT NULL %> " + getName() + " DROP");
+    getUnitTests().add(addTest.toString());
+
+    StringBuilder minusTest = new StringBuilder();
+    minusTest.append("[ 1 100 <% %> FOR ] [] [] [] [ 1 100 <% RAND %> FOR ] MAKEGTS 'a' STORE" + System.lineSeparator());
+    minusTest.append("[ 1 100 <% %> FOR ] [] [] [] [ 1 100 <% RAND %> FOR ] MAKEGTS 'b' STORE" + System.lineSeparator());
+    minusTest.append("$a $b <% - %> " + getName() + System.lineSeparator());
+    minusTest.append("$a $b - " + System.lineSeparator());
+    minusTest.append("<% == ASSERT NULL %> " + getName() + " DROP");
+    getUnitTests().add(minusTest.toString());
+
+    StringBuilder mulTest = new StringBuilder();
+    mulTest.append("[ 1 100 <% %> FOR ] [] [] [] [ 1 100 <% RAND %> FOR ] MAKEGTS 'a' STORE" + System.lineSeparator());
+    mulTest.append("[ 1 100 <% %> FOR ] [] [] [] [ 1 100 <% RAND %> FOR ] MAKEGTS 'b' STORE" + System.lineSeparator());
+    mulTest.append("$a $b <% + %> " + getName() + System.lineSeparator());
+    mulTest.append("$a $b + " + System.lineSeparator());
+    mulTest.append("<% == ASSERT NULL %> " + getName() + " DROP");
+    getUnitTests().add(mulTest.toString());
+  }
+
+  @Override
+  protected WarpScriptStack apply(Map<String, Object> formattedArgs, final WarpScriptStack stack) throws WarpScriptException {
+
+    //
+    // Retrieve arguments
+    //
+
+    GeoTimeSerie gts1 = (GeoTimeSerie) formattedArgs.get(GTS_1);
+    GeoTimeSerie gts2 = (GeoTimeSerie) formattedArgs.get(GTS_2);
+    final WarpScriptStack.Macro macro = (WarpScriptStack.Macro) formattedArgs.get(MACRO);
+
+    //
+    // Build BinaryOp
+    //
+
+    final WarpScriptStack stack_ = stack;
+
+    GTSOpsHelper.GTSBinaryOp op = new GTSOpsHelper.GTSBinaryOp() {
+
+      @Override
+      public Object op(GeoTimeSerie gtsa, GeoTimeSerie gtsb, int idxa, int idxb) throws WarpScriptException {
+
+        stack_.push(GTSHelper.valueAtIndex(gtsa, idxa));
```

Comment:
A function as generic as this one aims to be should take into account ticks,location,elevation and value, not only value.